### PR TITLE
Add usage metrics headers to CLI requests

### PR DESCRIPTION
**Pull request by janelletavares**
## Description of change

Fixes https://github.com/meroxa/cli/issues/536

## Type of change

- [x]  New feature
- [ ]  Bug fix
- [x]  Refactor
- [ ]  Documentation

## How was this tested?

- [x]  Unit Tests
- [ ]  Tested in staging
- [ ]  Tested in minikube

## Demo

| before | after |
|--------|-------|
|
Screenshot from 2022-11-17 16-49-39

|
Screenshot from 2022-11-17 16-49-08
|

### CODE REVIEW

File: cmd/meroxa/global/client.go [hunk 1]
```diff
@@ -172,9 +173,31 @@ func NewClient() (meroxa.Client, error) {
 		}
 	}
 	options = append(options, meroxa.WithAccountUUID(Config.GetString(UserAccountUUID)))
+	options = append(options, meroxa.WithHeader("Meroxa-CLI", "1"))
+	options = append(options, meroxa.WithHeader("Meroxa-CLI-Version", Version))
+	if goVersion := getGoVersion(); goVersion != "" {
+		options = append(options, meroxa.WithHeader("Meroxa-CLI-Go-Version", goVersion))
```
Comment (nassor) [inline cmd/meroxa/global/client.go:179]: `Meroxa-CLI-Go-Version` -> `Meroxa-CLI-App-Version`

**Comment by nassor** [comment]
There are two layers for headers:
- For all CLI commands:
  - `Meroxa-CLI`: if the request is coming from the CLI
  - `Meroxa-CLI-Version`: version of the meroxa CLI
  
- When the CLI command is a turbine app:
  - `Meroxa-CLI-App`: if it is coming from App
  - `Meroxa-CLI-App-Lang`: language used for the turbine integration
  - `Meroxa-CLI-App-Lang-Version`: version of the library used for the turbine integration
  
We could maybe remove `Meroxa-CLI` & `Meroxa-CLI-App` Because we could verify if the other fields are set. Different from the document, what do you think?

File: cmd/meroxa/global/client.go (hunk 1 above)
Comment (janelletavares) [reply cmd/meroxa/global/client.go:179]: all version is somewhere else

**Comment by raulb** [comment]
@janelletavares Thank you for addressing https://github.com/meroxa/product/issues/614. Could you open a separate a pull-request that fixes it? This helps with our automated changelog (example: https://github.com/meroxa/cli/releases/tag/v2.8.1) and helps with reviews, rollbacks (if necessary). Thank you.

File: cmd/meroxa/global/client.go [hunk 2]
```diff
@@ -172,9 +173,31 @@ func NewClient() (meroxa.Client, error) {
 		}
 	}
 	options = append(options, meroxa.WithAccountUUID(Config.GetString(UserAccountUUID)))
+	options = append(options, meroxa.WithHeader("Meroxa-CLI", "1"))
```
Comment (nassor) [inline cmd/meroxa/global/client.go:176]: I updated the implementation plan, we can remove this: https://www.notion.so/Turbine-Data-Flow-Implementation-Plans-Phase-1-6d4780375ddc4ca99a5ce7a97200bb1e?d=b667c1685a2c47f081e6d6475474487a#1df09d39eac04b0b9bdaa06f2bf9c5dc

File: cmd/meroxa/global/client.go (hunk 1 above)
Comment (janelletavares) [reply cmd/meroxa/global/client.go:179]: this is the version of go that people are using to run the meroxa CLI.  it's not a turbine value. we could potentially use this value to determine if there's a minimum version of go that we need to check for before running any commands, etc.

Comment (nassor) [reply cmd/meroxa/global/client.go:179]: I see. Going to add that in the CLI PR as well.

File: cmd/meroxa/root/apps/apps.go [hunk 3]
```diff
@@ -59,3 +66,28 @@ func (*Apps) SubCommands() []*cobra.Command {
...
 }
+
+type addHeader interface {
+	AddHeader(key, value string)
+}
+
+func addTurbineHeaders(c addHeader, lang, version string) {
+	c.AddHeader("Meroxa-CLI-App-Lang", lang)
+	if lang == turbine.JavaScript {
+		version = fmt.Sprintf("%s:cli%s", version, turbine.TurbineJSVersion)
```
Comment (nassor) [inline cmd/meroxa/root/apps/apps.go:77]: What is these two values?

Comment (janelletavares) [reply cmd/meroxa/root/apps/apps.go:77]: the first is the turbine-js-framework version and the second is the turbine-js-cli version. they can drift.

**Review by nassor — APPROVED** [review]

**Review by nassor — APPROVED** [review]

**Pull request merged**